### ISSUE TITLE
Cannot reproduce graph optimization step with inline_functions

### ISSUE THREAD

**Issue by Juanlu001** [dask/dask#1989]
I tried to reproduce http://dask.pydata.org/en/stable/optimize.html#example but the `inline_functions` step does not work for me:

```
>>> dsk2
{'count1': (<method 'count' of 'str' objects>,
  'apple orange apple pear orange pear pear',
  'orange'),
 'count2': (<method 'count' of 'str' objects>,
  'apple orange apple pear orange pear pear',
  'apple'),
 'nwords': (<function len>,
  (<method 'split' of 'str' objects>,
   'apple orange apple pear orange pear pear')),
 'out1': (<function __main__.format_str>, 'count1', 'orange', 'nwords'),
 'out2': (<function __main__.format_str>, 'count2', 'apple', 'nwords'),
 'print1': (<function __main__.print_and_return>, 'out1'),
 'print2': (<function __main__.print_and_return>, 'out2'),
 'val1': 'orange',
 'val2': 'apple',
 'words': 'apple orange apple pear orange pear pear'}
>>> dependencies
{'count1': set(),
 'count2': set(),
 'nwords': set(),
 'print1': {'count1', 'nwords'},
 'print2': {'count2', 'nwords'},
 'val1': set(),
 'val2': set(),
 'words': set()}
>>> inline_functions(dsk2, [len, str.split], dependencies=dependencies)
{'count1': (<method 'count' of 'str' objects>,
  'apple orange apple pear orange pear pear',
  'orange'),
 'count2': (<method 'count' of 'str' objects>,
  'apple orange apple pear orange pear pear',
  'apple'),
 'nwords': (<function len>,
  (<method 'split' of 'str' objects>,
   'apple orange apple pear orange pear pear')),
 'out1': (<function __main__.format_str>, 'count1', 'orange', 'nwords'),
 'out2': (<function __main__.format_str>, 'count2', 'apple', 'nwords'),
 'print1': (<function __main__.print_and_return>, 'out1'),
 'print2': (<function __main__.print_and_return>, 'out2'),
 'val1': 'orange',
 'val2': 'apple',
 'words': 'apple orange apple pear orange pear pear'}
```

As it can be seen, the graph does not simplify.

**Comment by Juanlu001** [comment]
If I try this other think, it throws a `KeyError`:

```
>>> inline_functions(dsk2, [], [len, str.split], dependencies=dependencies)

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-61-fd0647c743bd> in <module>()
----> 1 inline_functions(dsk2, [], [len, str.split], dependencies=dependencies)

/home/jlcano/.miniconda3/envs/b2c36/lib/python3.6/site-packages/dask/optimize.py in inline_functions(dsk, output, fast_functions, inline_constants, dependencies)
    266     if keys:
    267         dsk = inline(dsk, keys, inline_constants=inline_constants,
--> 268                      dependencies=dependencies)
    269         for k in keys:
    270             del dsk[k]

/home/jlcano/.miniconda3/envs/b2c36/lib/python3.6/site-packages/dask/optimize.py in inline(dsk, keys, inline_constants, dependencies)
    218     for key, val in dsk.items():
    219         if key not in dsk2:
--> 220             for item in keys & dependencies[key]:
    221                 val = subs(val, item, keysubs[item])
    222             dsk2[key] = val

KeyError: 'out2'
```

**Comment by jcrist** [comment]
Thanks for the bug report, this was due to a change in api not getting propagated to the docs. Fixed by #1990.

Note that unless you're working directly with [graphs](http://dask.pydata.org/en/latest/spec.html) or *maybe* [`dask.delayed`](http://dask.pydata.org/en/latest/delayed.html), you shouldn't need to use the optimizations directly at all.

**Comment by Juanlu001** [comment]
> Note that unless you're working directly with graphs or maybe dask.delayed, you shouldn't need to use the optimizations directly at all.

Sure! Just trying to reproduce the examples for learning purposes :) Thanks for the quick fix!
